### PR TITLE
GEODE-7852: test SNI client agin a geode (2-node) cluster

### PR DESCRIPTION
**Pull request by Bill**
Automated tests for [GEODE-7852](https://issues.apache.org/jira/browse/GEODE-7852): this PR sets up a dockerized 2-node Geode cluster and uses groups to target content to one node or the other.

- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Comment by Bill** [comment]
All checks were green except rat. I had missed a license file. Just added it so I believe CI will be green on this now.

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIAcceptanceTest.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void connectToSNIProxyDocker() {
         .port(15443)
         .getExternalPort();
     ClientCache cache = new ClientCacheFactory(gemFireProps)
-        .addPoolLocator("locator", 10334)
+        .addPoolLocator("locator-maeve", 10334)
```
Comment (echobravopapa) [inline geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIAcceptanceTest.java:90]: I would not have spelled that name correctly, kudos William

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/DualServerSNIAcceptanceTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+  }
+
+  @Test
+  public void dualServerTest() {
+    verifyPutAndGet("group-dolores", "region-dolores");
+  }
+
+  @Test
+  public void dualServerTest2() {
```
Comment (echobravopapa) [inline geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/DualServerSNIAcceptanceTest.java:100]: other than a reason to reference `clementine` what is `dualServerTest2` accomplishing?  
I wonder if a test that is meant to fail where we attempt to `cross the streams` would be a useful addition....

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIAcceptanceTest.java (hunk 1 above)
Comment (Bill) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIAcceptanceTest.java:90]: 🤠

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/DualServerSNIAcceptanceTest.java (hunk 2 above)
Comment (Bill) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/DualServerSNIAcceptanceTest.java]: ok @echobravopapa I added a couple "unreachability" tests in the latest commit

Comment (Bill) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/DualServerSNIAcceptanceTest.java]: test the test

**Comment by Bill** [comment]
I added comments @upthewaterspout. I had to rebase the branch so once the tests run I'll merge.

**Pull request merged**